metricflow/specs/where_filter_dimension.py

```diff
@@ -37,32 +38,48 @@ def __init__(  # noqa
         self._column_association_resolver = column_association_resolver
         self._name = name
         self._entity_path = entity_path
-        self.dimension_spec = self._dimension_spec_resolver.resolve_dimension_spec(name, entity_path)
-        self.time_dimension_spec: Optional[TimeDimensionSpec] = None
+        self.dimension_spec: DimensionSpec = self._dimension_spec_resolver.resolve_dimension_spec(
+            self._name, self._entity_path
+        )
+        self.date_part_name: Optional[str] = None
+        self.time_granularity_name: Optional[str] = None
+
+    @property
+    def time_dimension_spec(self) -> TimeDimensionSpec:
```

Comment:
Ah, so this returns a TimeDimension with DEFAULT_TIME_GRANULARITY if neither grain nor date_part are set. Interesting. I think that makes the most sense with where we're heading with time dimension expressions.